catalog/dags/data_refresh/create_filtered_index.py

```diff
@@ -0,0 +1,133 @@
+from airflow.utils.task_group import TaskGroup
+from airflow.utils.trigger_rule import TriggerRule
+
+from common import ingestion_server
+from common.constants import XCOM_PULL_TEMPLATE
+from data_refresh.data_refresh_types import DataRefresh
+
+
+def create_filtered_index_creation_task_groups(
+    data_refresh: DataRefresh,
+    origin_index_suffix: str | None,
+    destination_index_suffix: str | None,
+) -> tuple[TaskGroup, TaskGroup]:
+    """
+    Create the TaskGroups that performs filtered index creation and promotion for
+    the given DataRefresh.
+    """
+    media_type = data_refresh.media_type
+    target_alias = f"{media_type}-filtered"
+
+    def create_and_populate_filtered_index(
+        origin_index_suffix: str | None,
+        destination_index_suffix: str | None,
+    ):
+        create_payload = {}
+        if origin_index_suffix:
+            create_payload["origin_index_suffix"] = origin_index_suffix
+        if destination_index_suffix:
+            create_payload["destination_index_suffix"] = destination_index_suffix
+
+        return ingestion_server.trigger_and_wait_for_task(
+            action="CREATE_AND_POPULATE_FILTERED_INDEX",
+            model=media_type,
+            data=create_payload or None,
+            timeout=data_refresh.create_filtered_index_timeout,
+        )
+
+    def point_alias(destination_index_suffix: str) -> TaskGroup:
+        point_alias_payload = {
+            "alias": target_alias,
+            "index_suffix": f"{destination_index_suffix}-filtered",
+        }
+
+        with TaskGroup(group_id="point_alias") as point_alias_group:
+            ingestion_server.trigger_and_wait_for_task(
+                action="POINT_ALIAS",
+                model=media_type,
+                data=point_alias_payload,
+                timeout=timedelta(hours=12),  # matches the ingestion server's wait time
+            )
+        return point_alias_group
+
+    with TaskGroup(group_id="create_filtered_index") as create_filtered_index_group:
+        # If a destination index suffix isn't provided, we need to generate
+        # one so that we know where to point the alias
+        final_destination_index_suffix = (
+            ingestion_server.generate_index_suffix.override(
+                task_id="generate_filtered_index_suffix"
+            )(destination_index_suffix)
+        )
+
+        get_current_index_if_exists = ingestion_server.get_current_index(target_alias)
+
+        do_create, await_create = create_and_populate_filtered_index(
+            origin_index_suffix=origin_index_suffix,
+            destination_index_suffix=final_destination_index_suffix,
+        )
+
+        # Determine the destination index suffix and get the current index. The current
+        # index retrieval has to happen prior to any of the index creation steps to
+        # ensure the appropriate index information is retrieved.
+        [get_current_index_if_exists, destination_index_suffix]
```

Comment:
This step looks like a no-op, do you know if it's necessary?

Reply:
Totally, and in fact I'll move the task definitions around a little to be more straightforward while I'm at it.